src/Senders/CurlSender.php

```diff
@@ -84,4 +135,36 @@ public function setCurlOptions($handle, $scrubbedPayload, $accessToken)
             }
         }
     }
+
+    private function checkForCompletedRequests($accessToken)
+    {
+        do {
+            $curlResponse = curl_multi_exec($this->multiHandle, $active);
+        } while ($curlResponse == CURLM_CALL_MULTI_PERFORM);
+        while ($active && $curlResponse == CURLM_OK) {
```

Comment:
It's a bit confusing to read this code with 3 while loops. It might be a bit more concise if this was all put into the do/while loop with a switch statement to handle the different responses.